packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1142,38 +1374,34 @@ class TextInput {
   @visibleForTesting
   static void setChannel(MethodChannel newChannel) {
     assert(() {
-      _instance._channel = newChannel..setMethodCallHandler(_instance._handleTextInputInvocation);
+      _instance._channel = newChannel..setMethodCallHandler(
+          _TextInputChannelConnection._handleTextInputInvocation);
       return true;
     }());
   }
 
-  static final TextInput _instance = TextInput._();
-
-  static const List<TextInputAction> _androidSupportedInputActions = <TextInputAction>[
-    TextInputAction.none,
-    TextInputAction.unspecified,
-    TextInputAction.done,
-    TextInputAction.send,
-    TextInputAction.go,
-    TextInputAction.search,
-    TextInputAction.next,
-    TextInputAction.previous,
-    TextInputAction.newline,
-  ];
+  /// Sets the [TextInputSource] used to attach and detach text input clients.
```

Comment:
this sounds slightly confusing. "Sets the [TextInputSource] for [TextInputClient]s to attach to, when [attach] is called"?